patches/tModLoader/Terraria/GameContent/UI/States/UIEmotesMenu.cs.patch

```diff
@@ -0,0 +1,144 @@
+--- src/TerrariaNetCore/Terraria/GameContent/UI/States/UIEmotesMenu.cs
++++ src/tModLoader/Terraria/GameContent/UI/States/UIEmotesMenu.cs
+@@ -8,6 +_,7 @@
+ using Terraria.GameContent.UI.Elements;
+ using Terraria.ID;
+ using Terraria.Localization;
++using Terraria.ModLoader;
+ using Terraria.UI;
+ using Terraria.UI.Gamepad;
+ 
+@@ -22,6 +_,8 @@
+ 		private UIScrollbar _scrollBar;
+ 		private bool _isScrollbarAttached;
+ 
++		private int _totalGroups = 7; // Added by TML to support gamepad
++
+ 		public override void OnActivate() {
+ 			InitializePage();
+ 			if (Main.gameMenu) {
+@@ -86,6 +_,14 @@
+ 			TryAddingList(Language.GetText("UI.EmoteCategoryTownNPCs"), ref currentGroupIndex, 9, GetEmotesTownNPCs());
+ 			TryAddingList(Language.GetText("UI.EmoteCategoryCritters"), ref currentGroupIndex, 7, GetEmotesCritters());
+ 			TryAddingList(Language.GetText("UI.EmoteCategoryBosses"), ref currentGroupIndex, 8, GetEmotesBosses());
++
++			var modEmotes = EmoteBubbleLoader.GetAllUnlockedModEmotes();
++			foreach ((Mod mod, List<int> emotes) in modEmotes) {
++				// The maxEmotesPerRow parameter is actually useless
++				TryAddingList(new LocalizedText(mod.Name, mod.DisplayName), ref currentGroupIndex, -1, emotes);
++			}
++
++			_totalGroups = currentGroupIndex;
+ 		}
+ 
+ 		private void TryAddingList(LocalizedText title, ref int currentGroupIndex, int maxEmotesPerRow, List<int> emoteIds) {
+@@ -93,7 +_,7 @@
+ 				_list.Add(new EmotesGroupListItem(title, currentGroupIndex++, maxEmotesPerRow, emoteIds.ToArray()));
+ 		}
+ 
+-		private List<int> GetEmotesGeneral() {
```

Comment:
You could get better patches by editing the `TryAddingList` lines instead

Reply:
Editing the `TryAddingList` line will cause modders to get only vanilla emotes when they want to get a list of emotes for a category.